src/manifold/src/impl.h

```diff
@@ -76,6 +80,11 @@ struct Manifold::Impl {
   int NumVert() const { return vertPos_.size(); }
   int NumEdge() const { return halfedge_.size() / 2; }
   int NumTri() const { return halfedge_.size() / 3; }
+  int NumProp() const { return meshRelation_.numProp; }
+  int NumPropVert() const {
+    return NumProp() == 0 ? NumVert()
```

Comment:
Should we return `NumVert()` when there is no property or 0 as none of the vertices have any property? Actually I prefer having something like `Error(NoProperty)`/`Ok(n)`, although this isn't really popular in C++..

Reply:
I use `NumProp()` to determine if there aren't any properties, rather than `NumPropVert()`. I recall it simplified some code somewhere.

Reply:
OK thanks. Btw I will be a bit slow to review recently, so feel free to merge it first if you want, I will try to go over it later when I have more time. (plus this PR is huge and I am unfamiliar with the usecases, so will take more time to understand it)

Reply:
Thanks, and no problem. Yeah, this did turn out huge, since it involved a wholesale replacement of the barycentric part of `MeshRelation` when I realized how broken it was. The good news is I feel like the testing is a lot tighter now - this took forever because of how many errors were caught in those sanity checks. So I feel pretty confident about it algorithmically, though it could probably use further cleanup. As a bonus it seems to run a bit faster now too.